### PR TITLE
Verify Java imports are sorted

### PR DESCRIPTION
**Pull request by coverbeck**
**Description**

***Background***

The Maven impsort plugin sorts Java imports as part of the build. If you commit a Java file without having done a build,  imports may be out of order.

***Fix***
After a Maven build on CircleCI, check that there are no modified Java files in the Git tree. A modified Java file indicates that the impsort plugin only ran on CircleCI, and wasn't run before the file was committed.

For a CircleCI example of a failing build because imports weren't sorted, see [here](https://app.circleci.com/pipelines/github/dockstore/dockstore/11061/workflows/a0a76327-5740-4adc-9aab-f37ad29cef46/jobs/42108).

**Review Instructions**
1. Create a branch
2. Change the order of imports in any Java file
3. Do not do a Maven build
4. Commit & push
5. Look for an error in CircleCI.
6. Delete the branch

**Issue**
SEAB-3217

**Security and Privacy**

If there are any concerns that require extra attention from the security team, highlight them here and check the box when complete. 

- [x] Security and Privacy assessed

e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -249,6 +249,16 @@ jobs:
             # skip swagger validation because of difficult-to-fix operationId duplications
             # java -jar openapi-generator-cli.jar validate -i dockstore-webservice/src/main/resources/swagger.yaml
             java -jar openapi-generator-cli.jar validate -i dockstore-webservice/src/main/resources/openapi3/openapi.yaml
+      - run:
+          name: Check Java imports sorted
+          # A dirty Java file after a build probably means it was committed without building, and its imports are out of order
+          command: |
+            if [[ $(git diff --name-only | grep "\.java$") != '' ]]; then
```
Comment (coverbeck) [inline .circleci/config.yml:256]: To my surprise, although it makes sense when you think about it, there were other files showing up as modified -- the decrypted migrations files. So, look for modified Java files only.

**Review by kathy-t — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (denis-yuen) [reply .circleci/config.yml:256]: I forget, do we have a separate check for the openapi.yaml?
Does it make sense to put these two together?

**Review by denis-yuen — COMMENTED** [review]
Curious about story

**Comment by coverbeck** [comment]
> Curious about story

Can you please elaborate?

File: .circleci/config.yml (hunk 1 above)
Comment (coverbeck) [reply .circleci/config.yml:256]: Good catch, I added the check for openapi.yaml.

**Review by svonworl — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (denis-yuen) [reply .circleci/config.yml:256]: To be clear, I  don't remember the story behind this (i.e. I would have though we would have had it somewhere else before, but maybe we removed it), but worth a try

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**